pandas/tests/frame/test_sorting.py

```diff
@@ -116,9 +116,26 @@ def test_sort_values(self):
         self.assertRaises(ValueError, lambda: frame.sort_values(
             by=['A', 'B'], axis=2, inplace=True))
 
-        msg = 'When sorting by column, axis must be 0'
-        with assertRaisesRegexp(ValueError, msg):
-            frame.sort_values(by='A', axis=1)
+        # by row (axis=1): GH 10806
+        sorted_df = frame.sort_values(by=3, axis=1)
```

Comment:
Do you mean that the string `columns` should be supported as a valid `axis` value?  That's neat.  Can you point me to where that's documented though? I can't find reference to it.